platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collector.java

```diff
@@ -40,7 +40,10 @@ public interface Collector<T> extends ValueSupplier {
     /**
      * Returns a collector that never returns a missing value.
      */
-    default Collector<T> absentIgnoring() {
```

Comment:
> 🤔 There's no way out after you do `absentIgnoring(true)` (i.e. `absentIgnoring(false)` is a no op). IMO, the previous signature was a bit clearer, so we should keep `if (!ignoreAbsent)` outside. The same applies to `CollectionSupplier`.

Reply:
You are right. I added a convenience `absentIgnoringIfNeeded(boolean required)` as a default method, otherwise every call site would need to reason about whether to call `absentIgnoring()` or not.